# This configuration contains maps which define relationships of topics, nodes, launch files etc
# Most important property is a "group" which mostly corresponds to a launch file that can contain multiple nodes
# Therefore there are the following relationships:
# - Each node maps to one, and only one, group
# - Each topic maps to one, and only one, group
# - Each group maps to one, and only one, kill node
# - Each group maps to one, and only one, launch file

map_node_to_group: [
    {node: /base/laser, group: base},
    {node: /base/laser/laser_filter, group: base},
    {node: /base/neo_kinematics_differential_node, group: base},
    {node: /base/relayboard_v2/relayboard_v2_node, group: base},
    {node: /base/us_avoid_collision, group: base},
    {node: /base/velocity_smoother, group: base},
    {node: /battery_monitoring, group: battery_monitoring},
    {node: /cam_back/realsense2_camera_manager, group: cams},
    {node: /cam_back/republish_rgbd_sync_velo_back_rgb, group: cams},
    {node: /cam_back/rgbd_sync_back, group: cams},
    {node: /cam_back/rotate_image_depth, group: cams},
    {node: /cam_back/rotate_image_rgb, group: cams},
    {node: /cam_front/realsense2_camera_manager, group: cams},
    {node: /cam_front/republish_rgbd_sync_velo_front_rgb, group: cams},
    {node: /cam_front/rgbd_sync_front, group: cams},
    {node: /sobi/speech/conv_control, group: speech},
    {node: /sobi/speech/nlp, group: speech},
    {node: /sobi/speech/respeaker_node, group: speech},
    {node: /sobi/speech/stt_action, group: speech},
    {node: /sobi/speech/tts_action, group: speech},
    {node: /sobi/led_strips, group: led},
    {node: /sobi/led_panel, group: led},
    {node: /sobi/controller_spawner, group: ears},
    {node: /sobi/ear_hw_node, group: ears},
    {node: /sobi/message_node, group: message_node},
    {node: /rtabmap/rtabmap, group: rtabmap},
    {node: /move_base, group: move_base},
    {node: /cmr_lidarloop/lidar_loopdetection, group: rtabmap},
    {node: /cmr_lidarloop/lidar_loopdetector_server, group: rtabmap},
    {node: /cmr_lidarloop/lidar_registration_server, group: rtabmap},
    {node: /ekf, group: _},
    {node: /pcl_repub_velo, group: velodyne},
    {node: /pointcloud_to_laserscan_node, group: velodyne},
    {node: /robot_state_publisher, group: _},
    {node: /velodyne_nodelet_manager, group: velodyne},
    {node: /velodyne_nodelet_manager_driver, group: velodyne},
    {node: /velodyne_nodelet_manager_transform, group: velodyne},
    {node: /xsens_driver, group: _},
    {node: /spencer/perception_internal/people_tracking/srl_nearest_neighbor_tracker, group: people_perception},
    {node: /darknet_ros_front, group: people_perception},
    {node: /object3d_detector_fast, group: people_perception},
    {node: /spencer/perception_internal/detected_person_association/fuse_lasers_and_velo, group: people_perception},
    {node: /spencer/perception_internal/detected_person_association/convert_final_composite_detections_to_detections, group: people_perception},
    {node: /topological_map_manager, group: topological_navigation},
    {node: /topological_navigation, group: topological_navigation},
    {node: /topological_localisation, group: topological_navigation},
    {node: /roslaunch_dock, group: docking},
    {node: /undock, group: docking},
    {node: /no_mask_detection, group: no_mask_detection},

]
map_group_to_kill: [
    {group: base, kill_node: /platform_shutdown},
    {group: battery_monitoring, kill_node: /battery_monitoring_shutdown},
    {group: rtabmap, kill_node: /rtabmap_shutdown},
    {group: move_base, kill_node: /move_base_shutdown},
    {group: speech, kill_node: /speech_shutdown},
    {group: led, kill_node: /led_shutdown},
    {group: ears, kill_node: /ears_shutdown},
    {group: message_node, kill_node: /message_node_shutdown},
    {group: cams, kill_node: /cam_shutdown},
    {group: xsens_driver, kill_node: /imu_shutdown},
    {group: ekf, kill_node: /ekf_shutdown},
    {group: velodyne, kill_node: /velodyne_shutdown},
    {group: people_perception, kill_node: /people_perception_shutdown},
    {group: topological_navigation, kill_node: /topological_navigation_shutdown},
    {group: docking, kill_node: /dock_shutdown},
    {group: no_mask_detection, kill_node: /no_mask_detection_shutdown},
]
map_topic_to_group: [
    {topic: cmd_vel_safe, group: base},
    {topic: base, group: base},
    {topic: imu, group: xsens_driver},
    {topic: odom, group: ekf},
    {topic: odometry, group: ekf}
]
map_group_to_launch: [
    {group: base, pkg: cmr_driver, launch_file: platform_s.launch},
    {group: battery_monitoring, pkg: cmr_driver, launch_file: battery_monitoring_s.launch},
    {group: rtabmap, pkg: cmr_localization, launch_file: localization_s.launch},
    {group: move_base, pkg: cmr_navigation, launch_file: move_base_s.launch},
    {group: speech, pkg: cmr_driver, launch_file: voice_interface_s.launch},
    {group: led, pkg: cmr_driver, launch_file: led_s.launch},
    {group: ears, pkg: cmr_driver, launch_file: ears_s.launch},
    {group: message_node, pkg: cmr_api, launch_file: message_node.launch},
    {group: cams, pkg: cmr_driver, launch_file: cameras_s.launch},
    {group: rgbd_sync_front, pkg: cmr_driver, launch_file: cameras_s.launch},
    {group: rgbd_sync_back, pkg: cmr_driver, launch_file: cameras_s.launch},
    {group: xsens_driver, pkg: cmr_driver, launch_file: imu_s.launch},
    {group: ekf, pkg: cmr_driver, launch_file: ekf_s.launch},
    {group: velodyne, pkg: cmr_driver, launch_file: velodyne_s.launch},
    {group: people_perception, pkg: cmr_people_perception, launch_file: people_perception.launch},
    {group: robot_state_publisher, pkg: cmr_driver, launch_file: urdf_s.launch},
    {group: topological_navigation, pkg: cmr_navigation, launch_file: topological_navigation_s.launch},
    {group: docking, pkg: roslaunch_dock, launch_file: roslaunch_dock.launch},
    {group: no_mask_detection, pkg: cmr_no_mask_detection, launch_file: no_mask_detection.launch},

]
map_launch_to_delay: [
    {launch_file: platform_s.launch, delay: 20},
    {launch_file: battery_monitoring_s.launch, delay: 14},
    {launch_file: voice_interface_s.launch, delay: 25},
    {launch_file: localization_s.launch, delay: 70},
    {launch_file: move_base_s.launch, delay: 70},
    {launch_file: led_s.launch, delay: 12},
    {launch_file: ears_s.launch, delay: 12},
    {launch_file: message_node.launch, delay: 6},
    {launch_file: cameras_s.launch, delay: 60},
    {launch_file: rgbd_sync_s.launch, delay: 6},
    {launch_file: imu_s.launch, delay: 18},
    {launch_file: ekf_s.launch, delay: 18},
    {launch_file: velodyne_s.launch, delay: 30},
    {launch_file: people_perception.launch, delay: 30},
    {launch_file: urdf_s.launch, delay: 15},
    {launch_file: topological_navigation_s.launch, delay: 30},
    {launch_file: roslaunch_dock.launch, delay: 20},
    {launch_file: no_mask_detection.launch, delay: 15},
]
